File: c/isogram/src/isogram.c
#include "isogram.h"
#include <string.h>
#include <ctype.h>
#include <stdio.h>

bool is_isogram(const char phrase[])
{
    if (phrase == NULL) {
        return 0;
    }

    int map[26] = {0};
    int index = 0;
    int charIndex = 0;
    while(phrase[index] != '\0') {
        charIndex = tolower(phrase[index]) - 'a';
        map[charIndex]++;
        if (map[charIndex] > 1 && phrase[index] != ' ') {
            return 0;
        }
        index++;
    }
       
    return 1;
}

File: csharp/isogram/Isogram.cs
using System;
using System.Linq;

public static class Isogram
{
    // 这个解法，似乎理解错题目了，- -，英文的这些真是。。。
    public static bool IsIsogram2(string word)
    {
        word = word.ToLower();
        if (!word.Contains("-") && !word.Contains(" "))
        {
            return word.ToCharArray().Distinct().Count() == word.ToCharArray().Count();
        }
        else
        {
            foreach (var w in word.Split(new char[] { ' ', '-' }))
            {
                if (w.ToCharArray().Distinct().Count() != w.ToCharArray().Count())
                {
                    return false;
                }
            }

            return true;
        }
    }

    public static bool IsIsogram(string word)
    {
        var wordArr = word.ToLower().Replace(" ", "").Replace("-", "").ToCharArray();
        return wordArr.Distinct().Count() == wordArr.Count();
    }
}
